components/AppLayout/AppHeader.js

```diff
@@ -1,64 +1,281 @@
-import React from 'react';
+import React, { useState, useCallback } from 'react';
 import { t } from 'ttag';
+import cx from 'clsx';
+import NavLink from 'components/NavLink';
+import GlobalSearch from './GlobalSearch';
+import { grey } from '@material-ui/core/colors';
+import { makeStyles } from '@material-ui/core/styles';
+import MoreHorizIcon from '@material-ui/icons/MoreHoriz';
+import {
+  Button,
+  Menu,
+  MenuItem,
+  ListItemIcon,
+  Divider,
+  Typography,
+} from '@material-ui/core';
+import AccountCircleOutlinedIcon from '@material-ui/icons/AccountCircleOutlined';
+import ExitToAppRoundedIcon from '@material-ui/icons/ExitToAppRounded';
+import useMediaQuery from '@material-ui/core/useMediaQuery';
+import InfoIcon from '@material-ui/icons/Info';
+import { NAVBAR_HEIGHT, TABS_HEIGHT } from 'constants/size';
 import { EDITOR_FACEBOOK_GROUP, PROJECT_HACKFOLDR } from 'constants/urls';
-import Link from 'next/link';
-import UserName from './UserName';
+import getGravatar from 'lib/getGravatar';
 
-function AppHeader() {
-  return (
-    <header className="root">
-      <a className="logo hidden-xs" href="/">
-        <h1>{t`Cofacts`}</h1>
-      </a>
-      <nav className="nav">
-        <Link href="/articles">
-          <a className="nav-item">{t`Collected Messages`}</a>
-        </Link>
-        <Link href="/replies">
-          <a className="nav-item">{t`Replies`}</a>
-        </Link>
-        <a
-          href={EDITOR_FACEBOOK_GROUP}
-          target="_blank"
-          rel="noopener noreferrer"
-          className="nav-item hidden-xs"
-        >
-          {t`Editor forum`}
-        </a>
-        <a
-          href={PROJECT_HACKFOLDR}
-          target="_blank"
-          rel="noopener noreferrer"
-          className="nav-item hidden-xs"
-        >
-          {t`About`}
-        </a>
-      </nav>
-      <UserName />
-      <style jsx>{`
-        .root {
-          display: flex;
-          align-items: center;
-          padding: 0 24px;
-        }
-        .logo {
-          margin-right: 16px;
-        }
-        .nav {
-          margin-right: auto;
-          display: flex;
-        }
-        .nav-item {
-          padding: 8px;
-          border-left: 1px dashed #ccc;
-        }
+const MENU_BUTTON_WIDTH = 48;
+
+const useStyles = makeStyles({
+  root: {
+    position: 'sticky',
+    height: NAVBAR_HEIGHT + TABS_HEIGHT,
+    top: 0,
+    zIndex: 10,
+    '@media(min-width: 992px)': {
```

Comment:
We can [customize breakpoint values](https://material-ui.com/customization/default-theme/?expand-path=$.breakpoints.values) (maybe change `md` to `992`) so that we can use `theme.breakpoints.up('md')` here instead.

@LucienLee what do you think about changing [Material UI's default breakpoints](https://material-ui.com/customization/breakpoints/#breakpoints)? Should we bring `md` up to `992` or bring `lg` down to `992`?

Reply:
992 is defined by bootstrap, which is different from material-ui, using breakpoints defined by material-ui should be preferred, I'll fix this.

Reply:
Yeah, but I remember @LucienLee is referring to bootstrap breakpoints in his design. Not sure which breakpoint setup we should follow.